app/src/main/java/com/asfoundation/wallet/repository/AllowanceService.java

```diff
@@ -0,0 +1,63 @@
+package com.asfoundation.wallet.repository;
+
+import io.reactivex.Single;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.web3j.abi.FunctionEncoder;
+import org.web3j.abi.FunctionReturnDecoder;
+import org.web3j.abi.TypeReference;
+import org.web3j.abi.datatypes.Address;
+import org.web3j.abi.datatypes.Function;
+import org.web3j.abi.datatypes.Type;
+import org.web3j.abi.datatypes.generated.Uint256;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.protocol.core.methods.request.Transaction;
+
+import static org.web3j.protocol.core.methods.request.Transaction.createEthCallTransaction;
+
+public class AllowanceService {
+
+  private static final String DECIMALS = "1000000000000000000";
```

Comment:
Should we use the information about the token (TokenInfo) itself? Since the information is defined in the configuration of the projecto according with the network being used.